js/style/parse_color.js

```diff
@@ -1,46 +1,40 @@
 'use strict';
 
-var parseCSSColor = require('csscolorparser').parseCSSColor;
+var parseColorString = require('csscolorparser').parseCSSColor;
 var util = require('../util/util');
+var StyleFunction = require('./style_function');
 
-var colorCache = {};
+var cache = {};
 
-function parseColor(input) {
+module.exports = function parseColor(input) {
 
-    if (colorCache[input]) {
-        return colorCache[input];
+    if (StyleFunction.isFunctionDefinition(input)) {
 
-    // RGBA array
-    } else if (Array.isArray(input)) {
-        return input;
-
-    // GL function
-    } else if (input && input.stops) {
         return util.extend({}, input, {
-            stops: input.stops.map(parseFunctionStopColor)
+            stops: input.stops.map(function(stop) {
+                return [stop[0], parseColor(stop[1])];
+            })
```

Comment:
I don't see how it improves readability — the previous way was more readable for me.